spec/unit/functions/step_spec.rb

```diff
@@ -0,0 +1,93 @@
+require 'puppet'
+require 'spec_helper'
+require 'puppet_spec/compiler'
+
+require 'shared_behaviours/iterative_functions'
+
+describe 'the step method' do
+  include PuppetSpec::Compiler
+
+  it 'raises an error when used against an unsupported type' do
```

Comment:
"raises an error when give an argument of unsupported type" (or even better), "of a type that cannot be iterated"